File: Source/Entities/BuildController.cs
using Celeste.Mod.Entities;
using Celeste.Mod.YaoiHelper.Handlers;
using Microsoft.Xna.Framework;
using Monocle;

namespace Celeste.Mod.YaoiHelper.Entities;

[CustomEntity($"{nameof(YaoiHelper)}/{nameof(BuildController)}")]
[Tracked(false)]
public sealed class BuildController : Entity {
	private readonly int tileLimit;
	private readonly bool unlimited;
	private readonly bool allowEntityMode;

	public BuildController(EntityData data, Vector2 offset) : base(data.Position + offset) {
		tileLimit = data.Int("tile_limit");
		unlimited = data.Bool("unlimited");
		allowEntityMode = data.Bool("allow_entity_mode");

	}

	public override void Awake(Scene scene) {
		base.Awake(scene);
		if (scene is not Level level) return;
		BuildHandler.TileLimit = tileLimit;
		BuildHandler.Unlimited = unlimited;
		BuildHandler.AllowEntityMode = allowEntityMode;
	}
}

File: Loenn/entities/BuildController.lua

local buildController = {}

buildController.name = "YaoiHelper/BuildController"
buildController.texture = "LoennSprites/Entities/build_controller"
buildController.depth = -10000
buildController.justification = {0.5, 0.5}
buildController.placements = {
    name = "main",
	data = {
		allow_entity_mode = true;
		unlimited = false;
		tile_limit = 10,
	}
}

return buildController
